src/runtime/virtcontainers/fc.go

```diff
@@ -435,28 +440,29 @@ func (fc *firecracker) fcEnd() (err error) {
 
 	pid := fc.info.PID
 
-	// Send a SIGTERM to the VM process to try to stop it properly
-	if err = syscall.Kill(pid, syscall.SIGTERM); err != nil {
-		if err == syscall.ESRCH {
-			return nil
+	// Trying to gracefully poweroff the VM
+	if archFcPowerOffFunc != nil {
+		if err := archFcPowerOffFunc(fc); err != nil {
+			return err
 		}
-		return err
-	}
 
-	// Wait for the VM process to terminate
-	tInit := time.Now()
-	for {
-		if err = syscall.Kill(pid, syscall.Signal(0)); err != nil {
-			return nil
-		}
+		// Wait for the VM process to terminate
+		tInit := time.Now()
+		for {
+			if err = syscall.Kill(pid, syscall.Signal(0)); err != nil {
```

Comment:
Nit: strictly you should check that `err` is `ESRCH` but since this is a tweak to the existing code, not essential to change.